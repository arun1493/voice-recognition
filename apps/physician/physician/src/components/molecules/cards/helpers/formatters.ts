import i18n from "i18next";
import _get from "lodash/get";
import { EMPTY_OBJECT } from "@suki-web/suki-business";

import {
  Allergy,
  AllergyReaction,
  CardMedicalHistoryProps,
  CardSurgicalHistoryProps,
  ParagraphCard,
  Problem
} from "../../../../types";
import { CardType } from "../../../../constants";

const formatParagraphCardData = (data: CardMedicalHistoryProps | CardSurgicalHistoryProps | any, cardType: string) => {
  const paragraphData: Array<ParagraphCard> = [];

  switch (cardType) {
    case CardType.ALLERGIES: {
      const { allergies, nkda } = data || EMPTY_OBJECT;

      allergies &&
        allergies.forEach((allergyItem: Allergy) => {
          const { allergen_name, reactions } = allergyItem || EMPTY_OBJECT;
          let allergyReactionContent = "";
          reactions &&
            reactions.forEach((allergyItem: AllergyReaction) => {
              const { reaction_name } = allergyItem;
              if (reaction_name) allergyReactionContent = allergyReactionContent.concat(reaction_name, ", ");
            });

          allergyReactionContent = allergyReactionContent.slice(0, -2); // to remove comma at the end of string

          if (allergen_name)
            paragraphData.push({
              title: allergen_name,
              content: allergyReactionContent
            });
        });

      if (nkda)
        paragraphData.push({
          content: i18n.t("pages.note.cards.errors.allergiesNkda")
        });
      break;
    }
    case CardType.PAST_MEDICAL_HISTORY: {
      const { active_problems } = data || EMPTY_OBJECT;
      active_problems &&
        active_problems.forEach((activeProblem: Problem) => {
          const description = _get(activeProblem, "note");
          const error = _get(activeProblem, "error");
          const title = _get(activeProblem, "diagnosis.icd_description");
          if (!error) paragraphData.push({ title, content: description });
        });
      break;
    }

    case CardType.PAST_SURGICAL_HISTORY: {
      data &&
        data.forEach((surgicalHistoryItem: Problem) => {
          const { description, note, error } = surgicalHistoryItem;
          if (!error) paragraphData.push({ title: description, content: note });
        });
      break;
    }

    default:
      break;
  }
  return paragraphData;
};

export { formatParagraphCardData };
