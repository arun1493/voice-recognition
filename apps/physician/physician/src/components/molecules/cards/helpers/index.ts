export * from "./labels";
export * from "./formatters";
