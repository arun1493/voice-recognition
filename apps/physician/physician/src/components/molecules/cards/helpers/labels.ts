import i18n from "i18next";

import { CardType, VITAL_SIGNS } from "../../../../constants";

const getCardTitle = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.titles.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.titles.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.titles.vitals");
    default:
      return "";
  }
};

const getVitalSign = (sign: string) => {
  switch (sign) {
    case VITAL_SIGNS.BLOOD_PRESSURE:
      return i18n.t("pages.note.cards.vitalSigns.bloodPressure");
    case VITAL_SIGNS.BMI:
      return i18n.t("pages.note.cards.vitalSigns.bmi");
    case VITAL_SIGNS.HEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.height");
    case VITAL_SIGNS.PULSE:
      return i18n.t("pages.note.cards.vitalSigns.pulse");
    case VITAL_SIGNS.WEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.weight");
    case VITAL_SIGNS.HEART_RATE:
      return i18n.t("pages.note.cards.vitalSigns.heartRate");
    case VITAL_SIGNS.RESPIRATORY_RATE:
      return i18n.t("pages.note.cards.vitalSigns.respiratoryRate");
    case VITAL_SIGNS.TEMPERATURE:
      return i18n.t("pages.note.cards.vitalSigns.temperature");
    default:
      return null;
  }
};

const getErrorMessage = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.errors.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.errors.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.errors.vitals");
    default:
      return "";
  }
};

export { getCardTitle, getErrorMessage, getVitalSign };
