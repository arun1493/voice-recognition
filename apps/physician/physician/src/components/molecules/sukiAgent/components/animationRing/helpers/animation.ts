import { RING_ANIMATION_DURATION, RING_BASE_OPACITY, RING_SCALE, SukiAgentState } from "../../../../../../constants";

function isValidSukiAgentStateForSingleRingAnimation(agentStatus: SukiAgentState, isAgentActive: boolean) {
  return (isAgentActive && agentStatus === SukiAgentState.AWAKE) || agentStatus === SukiAgentState.PROCESSING;
}

function isValidSukiAgentStateForMultipleRingAnimation(agentStatus: SukiAgentState, isAgentActive: boolean) {
  return (
    isAgentActive &&
    !isValidSukiAgentStateForSingleRingAnimation(agentStatus, isAgentActive) &&
    agentStatus === SukiAgentState.LISTENING
  );
}

function getDictationRingVariants(ringsCount: number) {
  const dictationRingVariants = [];
  for (let i = ringsCount; i > 0; i--) {
    const RING_VARIANTS = {
      oscillate: {
        opacity: RING_BASE_OPACITY - i * 0.2,
        scale: RING_SCALE + i * 0.01,
        transition: {
          yoyo: Infinity,
          duration: RING_ANIMATION_DURATION,
          type: "tween",
          delay: i * 0.1,
          ease: "easeInOut"
        }
      }
    };

    dictationRingVariants.push(RING_VARIANTS);
  }
  return dictationRingVariants;
}

export {
  getDictationRingVariants,
  isValidSukiAgentStateForMultipleRingAnimation,
  isValidSukiAgentStateForSingleRingAnimation
};
