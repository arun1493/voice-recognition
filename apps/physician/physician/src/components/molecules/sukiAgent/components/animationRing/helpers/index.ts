export * from "./ringSize";
export * from "./animation";
