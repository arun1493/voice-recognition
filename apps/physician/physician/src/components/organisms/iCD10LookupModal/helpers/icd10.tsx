import _isEmpty from "lodash/isEmpty";

const formatText = (text: string, highlight: string) => {
  if (_isEmpty(text)) return;
  const parts = text.split(new RegExp(`(${highlight})`, "gi"));
  return parts.map((part) => (part.toLowerCase() === highlight.toLowerCase() ? <strong>{part}</strong> : part));
};

const formatCode = (code: string) => {
  if (code?.length > 3) {
    return code?.substring(0, 3) + "." + code?.substring(3);
  }
  return code;
};

export { formatCode, formatText };
