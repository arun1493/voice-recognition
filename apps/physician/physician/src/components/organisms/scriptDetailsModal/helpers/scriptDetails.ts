import { ScreenSize } from "@suki-web/suki-ui";

const { SM, XS, XXS } = ScreenSize;

const getWidth = (breakPoint: ScreenSize) => {
  switch (breakPoint) {
    case XXS:
      return "100vw";
    case XS:
      return 484;
    case SM:
      return 664;
    default:
      return 782;
  }
};

const getHeight = (breakPoint: ScreenSize) =>
  breakPoint === XS || breakPoint === XXS ? "100vh" : "calc( 100vh - 56px )";

const getMaxHeight = (breakPoint: ScreenSize) => (breakPoint === XS || breakPoint === XXS ? "100vh" : 0);

export { getHeight, getMaxHeight, getWidth };
