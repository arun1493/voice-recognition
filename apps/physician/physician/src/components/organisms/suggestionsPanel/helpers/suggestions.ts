import i18n from "i18next";

import { SuggestionTypes } from "../../../../constants";

const getTextBySuggestion = (suggestion: string) => {
  switch (suggestion) {
    case SuggestionTypes.PL:
      return i18n.t("components.suggestionsPanel.pl");
    case SuggestionTypes.ED:
      return i18n.t("components.suggestionsPanel.ed");
    default:
      return "";
  }
};

export { getTextBySuggestion };
