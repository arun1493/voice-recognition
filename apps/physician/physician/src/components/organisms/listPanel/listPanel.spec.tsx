import { format } from "date-fns";
import "@testing-library/jest-dom/extend-expect";
import { NoteStatus, PatientModes } from "@suki-web/suki-ui";
import { cleanup, RenderResult } from "@testing-library/react";

import { configureI18n } from "../../../helpers";
import { ListPanel, ListPanelProps } from "./index";
import { componentSetupWithStore } from "../../../hocs";
import { shouldMatchSnapshot } from "../../../helpers/snapshot.spec";

const props: ListPanelProps = {
  age: "29",
  fullName: "John Snow",
  gender: "M",
  status: NoteStatus.USER_IS_DONE,
  startsAt: "Tue Jul 12 2022 14:07:39",
  type: PatientModes.SCHEDULE
};

const setup = componentSetupWithStore(() => <ListPanel {...props} />);

describe("Patient List Panel", () => {
  let component: RenderResult;

  beforeEach(() => {
    configureI18n();
    component = setup();
  });

  afterEach(() => {
    cleanup();
  });

  it("should render successfully", () => {
    const { baseElement } = component;
    expect(baseElement).toBeTruthy();
  });

  it("should render schedule type details properly", () => {
    const { baseElement } = component;
    const patientName = baseElement.querySelector(`[data-cy="patient-name"]`)?.textContent;
    const patientAge = baseElement.querySelector(`[data-cy="patient-age"]`)?.textContent;
    const patientGender = baseElement.querySelector(`[data-cy="patient-gender"]`)?.textContent;
    const patientStartsAt = baseElement.querySelector(`[data-cy="patient-appointment-time"]`)?.textContent;

    expect(props.fullName).toEqual(patientName);
    expect(props.age).toEqual(patientAge);
    expect(props.gender).toEqual(patientGender);
    expect(format(new Date(props.startsAt as string), "p")).toEqual(patientStartsAt);
  });

  it("should match snapshot", () => {
    shouldMatchSnapshot(ListPanel, props);
  });
});
