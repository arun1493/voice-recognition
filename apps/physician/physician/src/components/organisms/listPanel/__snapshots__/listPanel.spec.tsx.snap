// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Patient List Panel should match snapshot 1`] = `
<div
  className="listPanelstyled__StyledBody-sc-1mhlj0f-0 iytHHV"
  data-cy=""
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="patientCellstyled__PatientCellContainer-sc-1r066ns-0 bwkvmv"
    type="schedule"
  >
    <p
      className="textstyled__StyledText-sc-1vedkxi-0 fytJms patientCellstyled__PatientName-sc-1r066ns-1 bjFiTR"
      color="inherit"
      cursor="pointer"
      data-cy="patient-name"
      onClick={[Function]}
      size="md"
    >
      John Snow
    </p>
    <span
      className="patientCellstyled__PatientAge-sc-1r066ns-3 cJQZsw"
      data-cy="patient-age"
    >
      29
    </span>
    <span
      className="patientCellstyled__PatientGender-sc-1r066ns-4 btTaoo"
      data-cy="patient-gender"
    >
      M
    </span>
  </div>
  <div
    className="listPanelstyled__StyledTime-sc-1mhlj0f-1 iUgSsb"
  >
    <p
      className="textstyled__StyledText-sc-1vedkxi-0 ixFoFi"
      color="inherit"
      cursor="pointer"
      data-cy="patient-appointment-time"
      onClick={[Function]}
      size="sm"
    >
      2:07 PM
    </p>
  </div>
  <div
    className="statusCellstyled__StyledContainer-sc-3olh8y-1 eBriFQ"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <svg
      fill="none"
      height={24}
      viewBox="0 0 21 23"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18.3851 3.58965L17.6851 3.08845C17.5735 3.00861 17.4605 2.93077 17.3459 2.85497C17.198 2.7571 17.0476 2.66265 16.8948 2.57172C15.2172 1.57323 13.2516 0.998535 11.1468 0.998535C4.99061 0.998535 0 5.9234 0 11.9985C0 18.0737 4.99061 22.9985 11.1468 22.9985C15.2066 22.9985 18.7605 20.8574 20.7108 17.6523C20.8979 17.3447 20.8746 17.0405 20.7071 16.7726C20.5276 16.4856 20.1741 16.2422 19.7163 16.2012C19.4754 16.1796 19.16 16.3183 18.9476 16.6634C17.35 19.2601 14.4546 20.9985 11.1468 20.9985C6.10992 20.9985 2.0267 16.9691 2.0267 11.9985C2.0267 7.02797 6.10992 2.99854 11.1468 2.99854C12.4543 2.99854 13.7 3.27065 14.8269 3.76157C15.1934 3.92123 15.5471 4.10393 15.8862 4.30779L16.9868 4.96955L18.3851 3.58965Z"
        fill="#999999"
      />
      <path
        d="M9.26316 7.9459C9.26316 7.30642 9.78157 6.78801 10.4211 6.78801C11.0605 6.78801 11.5789 7.30642 11.5789 7.9459V13.7354C11.5789 14.3749 11.0605 14.8933 10.4211 14.8933C9.78157 14.8933 9.26316 14.3749 9.26316 13.7354V7.9459Z"
        fill="#999999"
      />
      <path
        d="M13.8947 12.5775C14.5342 12.5775 15.0526 13.0959 15.0526 13.7354C15.0526 14.3749 14.5342 14.8933 13.8947 14.8933H10.4211C9.78157 14.8933 9.26316 14.3749 9.26316 13.7354C9.26316 13.0959 9.78157 12.5775 10.4211 12.5775H13.8947Z"
        fill="#999999"
      />
    </svg>
  </div>
</div>
`;
